src/archemist/core/optimisation/recipe_generator.py

```diff
@@ -0,0 +1,94 @@
+import pandas as pd
+import yaml
+from archemist.core.persistence.yaml_handler import YamlHandler
+from nested_lookup import nested_update
+from pathlib import Path
+import logging
+import re
+from typing import Dict
+
+class RecipeGenerator:
+    def __init__(self, template_path, recipe_path) -> None:
+        self._recipe_dict  = YamlHandler.load_recipe_file(template_path)
+        self._new_recipe_path = recipe_path
+        
+        self._placeholder_keys_all = {}
+        self._placeholder_keys_any = {}
+        self._find_placeholders(self._recipe_dict)
+
+    # Methods
+    def generate_recipe(self, optimised_parameters_data, file_name):
```

Comment:
when you generate a recipe you need to give it a new id, otherwise, the recipe watchdog won't pick it up